src/runtime_src/core/edge/skd/main.cpp

```diff
@@ -34,65 +38,85 @@
  * and exit.
  */
 
+using severity_level = xrt_core::message::severity_level;
+
 int main(int argc, char *argv[])
 {
-  pid_t pid, sid;
-  xclDeviceHandle handle;
-  xclSKCmd cmd;
-
-  pid = fork();
-  if (pid < 0) {
+  pid_t pid = fork();
+  if (pid < 0)
     exit(EXIT_FAILURE);
-  }
+
   if (pid > 0)
     exit(EXIT_SUCCESS);
 
+  // Starting the child process
   umask(0);
 
-  /* Open syslog and send the first message */
-  openlog("skd", LOG_PID | LOG_CONS, LOG_LOCAL0);
-  syslog(LOG_INFO, "Daemon Start...\n");
+  // Send the first message to XRT log
+  const std::string msg = "Daemon Start...";
+  xrt_core::message::send(severity_level::info, "SKD", msg);
 
-  /* Create a new SID for the child process */
-  sid = setsid();
+  // Create a new SID for the child process
+  pid_t sid = setsid();
   if (sid < 0) {
-    syslog(LOG_INFO, "Set SID failed. Daemon exiting\n");
-    closelog();
+    const auto errMsg = "Set SID failed. Daemon exiting";
+    xrt_core::message::send(severity_level::error, "SKD", errMsg);
     exit(EXIT_FAILURE);
   }
-  syslog(LOG_INFO, "SID set %d\n", sid);
+  const auto sid_msg = boost::format("SID set %d") % sid;
+  xrt_core::message::send(severity_level::info, "SKD", sid_msg.str());
 
   if (chdir("/") < 0) {
-    syslog(LOG_INFO, "Could NOT change to \"/\" directory\n");
-    closelog();
+    const auto errMsg = "Could NOT change to \"/\" directory";
+    xrt_core::message::send(severity_level::error, "SKD", errMsg);
     exit(EXIT_FAILURE);
```

Comment:
Report the errno for chdir failure.
Any idea why are we moving to the root directory? Should we have our own dedicated run directory?